src/App.tsx

```diff
@@ -0,0 +1,52 @@
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import React, { useState, createContext } from "react";
+import "./App.css";
+import CartList from "./Pages/CartList";
+import HomePage from "./Pages/HomePage";
+import ProductPage from "./Pages/ProductPage";
+import NotFoundPage from "./Pages/NotFoundPage";
+
+interface Product {
+  id: number;
+  image: string;
+  description: string;
+  name: string;
+  price: number;
+  amount: number;
+}
+
+const router = createBrowserRouter([
```

Comment:
dont use this. use this - import { Route, Routes, useNavigate } from 'react-router-dom';